### PR TITLE
[CP] Katello 3.12.2 Cherrypicks from Toolbelt

### PR DESCRIPTION
**Pull request by chris1984**
@jturel let me know how this looks?

The only one missing according to toolbelt is:

https://github.com/Katello/katello/pull/8238#event-2519836202

### CODE REVIEW

**Comment by jturel** [comment]
[test katello]

**Comment by jturel** [comment]
Looking good so far. Let me know when all the picks are here

**Comment by chris1984** [comment]
[test katello]

**Comment by chris1984** [comment]
@jlsherrill Hey I was looking through the cherry-picks and see https://github.com/Katello/katello/pull/8243/commits/26b714543d1349b64b4e5ace214344d78a9e3556 has failing tests. 

https://ci.theforeman.org/blue/organizations/jenkins/katello-pr-test/detail/katello-pr-test/1332/tests/

https://github.com/Katello/katello/pull/8243/commits/26b714543d1349b64b4e5ace214344d78a9e3556#diff-4fe5196cbfee7520a01c24810b159298

I was talking to @jturel and he was saying this test will prob intermittently fail. When you get time can you provide a patch and I can bring it in?

**Comment by jlsherrill** [comment]
@chris1984 i think this should fix it:

```
diff --git a/test/actions/katello/repository/discover_test.rb b/test/actions/katello/repository/discover_test.rb
index 7abe7faf4e..d7f8b1a055 100644
--- a/test/actions/katello/repository/discover_test.rb
+++ b/test/actions/katello/repository/discover_test.rb
@@ -13,6 +13,7 @@ module Actions
     let(:action_class) { ::Actions::Katello::Repository::Discover }
 
     def setup
+      SETTINGS[:katello][:cdn_proxy] = {}
       get_organization #ensure we have an org label
     end
 
```

**Comment by chris1984** [comment]
@jturel Katello tests are passing now, it looks like the assets-precompile is failing with a Java out of memory error? https://ci.theforeman.org/blue/organizations/jenkins/katello-pr-test/detail/katello-pr-test/1344/pipeline/69#step-125-log-612 and the new content view failing test, which was not there before a few days ago.

**Comment by jturel** [comment]
I guess we'll have to kick the tests again until they go green :smile: 

I think the repo discovery test you fixed will also be intermittent on master branch. Can you open a PR with a 'Refs #<justin's issue #> clear proxy settings for tests' so that we don't run into the same problem there?

**Comment by jturel** [comment]
[test katello]

**Comment by chris1984** [comment]
pr created here for test fix https://github.com/Katello/katello/pull/8253

**Comment by chris1984** [comment]
[test katello]

**Comment by jturel** [comment]
[test katello]

**Comment by jturel** [comment]
[test katello]

**Comment by chris1984** [comment]
[test katello]

**Comment by chris1984** [comment]
happending even when i reset back before the pr:

```bash
==== JS stack trace =========================================

Security context: 0x1975131cf781 <JS Object>
    1: /* anonymous */(aka /* anonymous */) [/home/vagrant/foreman/node_modules/webpack-sources/lib/applySourceMap.js:~58] [pc=0x1168d43e367a] (this=0x197513104381 <undefined>,chunk=0x2bd2038963f9 <String[2]: t,>,middleMapping=0x2efd26b8d401 <an Object with map 0x7cabdcad01>)
    2: walk(aka SourceNode_walk) [/home/vagrant/foreman/node_modules/source-map/lib/source-node.js:~221] [pc=0x1168ce6e8...

FATAL ERROR: CALL_AND_RETRY_LAST Allocation failed - JavaScript heap out of memory
 1: node::Abort() [node]
 2: 0x710fec [node]
 3: v8::Utils::ReportApiFailure(char const*, char const*) [node]
 4: v8::internal::V8::FatalProcessOutOfMemory(char const*, bool) [node]
 5: v8::internal::Factory::NewInternalizedStringImpl(v8::internal::Handle<v8::internal::String>, int, unsigned int) [node]
 6: v8::internal::InternalizedStringKey::AsHandle(v8::internal::Isolate*) [node]
 7: v8::internal::StringTable::LookupKey(v8::internal::Isolate*, v8::internal::HashTableKey*) [node]
 8: v8::internal::StringTable::LookupString(v8::internal::Isolate*, v8::internal::Handle<v8::internal::String>) [node]
 9: v8::internal::JSReceiver::HasProperty(v8::internal::Handle<v8::internal::JSReceiver>, v8::internal::Handle<v8::internal::Name>) [node]
10: v8::internal::Runtime_HasProperty(int, v8::internal::Object**, v8::internal::Isolate*) [node]
11: 0x1168ca6092a7
rake aborted!
Command failed with status (): [/home/vagrant/foreman/node_modules/webpack...]
```
```bash
[vagrant@centos7-katello-devel katello]$ git log
commit 218a28423ba8514aedb8adba087068a78401f9c9
Author: Chris Roberts <chrobert@redhat.com>
Date:   Wed Jul 31 09:48:54 2019 -0400

    Bump version to 3.12.2
```

**Comment by chris1984** [comment]
[test katello]

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,20 @@
+# 3.12.2 Iron Stout (2019-07-31)
+
+## Features
```
Comment (jturel) [inline CHANGELOG.md]: Empty features heading, otherwise LGTM! Sorry I didn't catch it before

**Comment by chris1984** [comment]
@jturel updated

**Pull request merged**